internal/extproc/translator/openai_awsbedrock.go

```diff
@@ -152,113 +180,194 @@ func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) openAIToolsToBedrockToolC
 	return nil
 }
 
-// OpenAIMessageToBedrockMessage converts openai ChatCompletion messages to aws bedrock messages
-func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) OpenAIMessageToBedrockMessage(openAIReq *openai.ChatCompletionRequest,
+// openAIMessageToBedrockMessageRoleUser converts openai user role message
+// returns a list of bedrock message and error
+// the tool content is appended to the list of bedrock message list if tool_call is in openai message
+func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) openAIMessageToBedrockMessageRoleUser(
+	openAiMessage *openai.ChatCompletionMessageParamUnion,
+) ([]*awsbedrock.Message, error) {
```

Comment:
this returned slice allocation is wasted as at the callsite the results are appended into another slice and this return value is discarded immediately